main.nf

```diff
@@ -165,60 +170,126 @@ process StringTie2 {
-ch_annot_feature_count
-    .concat(ch_txome_feature_count)
-    .set{ ch_feature_count }
+    output:
+    val "$stringtie_dir/merged.combined.gtf" into ch_merged_gtf
 
+    when:
+    transcriptquant == "stringtie"
 
-/*
- * STEP 3 - FeatureCounts
- */
- process FeatureCounts {
+    script:
+    """
+    ls -d -1 $PWD/$stringtie_dir/*.out.gtf > $PWD/$stringtie_dir/gtf_list.txt
+    echo "$annot" >> $PWD/$stringtie_dir/gtf_list.txt
+    gffcompare -i $PWD/$stringtie_dir/gtf_list.txt -o $PWD/$stringtie_dir/merged
+    gffcompare --version &> gffcompare.version
+    """
+}
+
+ch_txome_feature_count
+   .combine(ch_merged_gtf)
+   .set {ch_feature_count}
+
+process FeatureCounts {
      publishDir "${params.outdir}/featureCounts_transcript", mode: 'copy',
          saveAs: { filename ->
-                 if (!filename.endsWith(".version") && !filename.endsWith(".gene_counts.txt")) filename
-                 }
-     publishDir "${params.outdir}/featureCounts_gene", mode: 'copy',
-         saveAs: { filename ->
-                 if (!filename.endsWith(".version") && !filename.endsWith(".transcript_counts.txt")) filename
+                 if (!filename.endsWith(".version")) filename
                  }
 
      input:
-     set val(name), file(bam), file(annot) from ch_feature_count
+     set val(name), file(bam), val(annot) from ch_feature_count
+     val transcriptquant from ch_transcriptquant
 
      output:
      file("*.txt") into ch_counts
      file("*.version") into ch_feat_counts_version
-     val "result/featureCounts_gene" into ch_deseq_indir
-     val "result/featureCounts_transcript" into ch_dex_indir
+     val "$baseDir/results/featureCounts_transcript" into ch_deseq2_indir
+     val "$baseDir/results/featureCounts_transcript" into ch_dexseq_indir
+
+     when:
+     transcriptquant == "stringtie"
 
      script:
-     txome_recon = (annot =~ /\.out\.gtf/) ? ".tx_recon" : ""
      """
-     featureCounts -T $task.cpus -a $annot -o ${name}${txome_recon}.gene_counts.txt $bam
-     featureCounts -g transcript_id --extraAttributes gene_id  -T $task.cpus -a $annot -o ${name}${txome_recon}.transcript_counts.txt $bam
+     featureCounts -g transcript_id --extraAttributes gene_id  -T $task.cpus -a $PWD/$annot -o ${name}.transcript_counts.txt $bam
      featureCounts -v &> featureCounts.version
      """
  }
 
 /*
- * STEP 4 - DESeq2
+ * STEP 2b - Bambu
+ */
+params.Bambuscript= "$baseDir/bin/runBambu.R"
+ch_Bambuscript = Channel.fromPath("$params.Bambuscript", checkIfExists:true)
+
+process Bambu {
+  publishDir "${params.outdir}/Bambu", mode: 'copy',
+        saveAs: { filename ->
+                      if (!filename.endsWith(".version")) filename
+                }
+
+  input:
+  file Bambuscript from ch_Bambuscript
+  file sampleinfo from ch_input
+  val transcriptquant from ch_transcriptquant
+
+  output:
+  val "$baseDir/results/Bambu/counts_gene.txt" into ch_deseq2_in
```

Comment:
should it $params.outdir instead? same for the line below

Reply:
I have gotten problems using $params.outdir instead of $baseDir/results, which $params.outdir maps to $baseDir/work/<random_string>/.